batch/pom.xml

```diff
@@ -273,6 +276,26 @@
       </exclusions>
     </dependency>
 
+    <!--JDBC specific dependencies-->
+    <dependency>
+      <!--JDBC connection pool will make concurrent JDBC connections more efficient-->
```

Comment:
Did not investigate this, however doing a simple search, it is quite clear HikariCP is 100 folds performant than other connection pools. We should definitely switch to Hikari! Thanks, @bashir2 for the suggestion!

Reply:
> Hi @bashir2
> 
> Please feel free to review this PR when you get the chance! I have addressed most of the suggestions including these key changes
> 
> 1. extracting bundles instead of resources
> 2. Done some major restructuring and refactoring as suggested
> 3. Use PTransformer classes to improve code readability and testing using [this guideline on how to package transformer](https://beam.apache.org/contribute/ptransform-style-guide/)
> 4. added a few more unit test
> 
> Lastly, I also tried using Hikari instead of c3po, however, I have been having compatibility issues. So APache beam PTransform/DoFn expects the datasource to be [serializable](https://beam.apache.org/releases/javadoc/2.7.0/org/apache/beam/sdk/io/jdbc/JdbcIO.html). I tried injecting Hikari using `@setup()` without success as advised [here](https://beam.apache.org/releases/javadoc/2.4.0/index.html?org/apache/beam/sdk/transforms/DoFn.Setup.html). Also tried using ` .withDataSourceProviderFn((SerializableFunction<Void, DataSource>)`. Most recommendation keeps on pointing back to c3po. Please let me know if you know a walk around for this issue. A

please see the above comment

Reply:
I see, thanks, let's leave the HikariCP issue for future (and thanks for creating the ticket for it).